hibernate-core/src/test/java/org/hibernate/test/enums/OrdinalEnumTypeTest.java

```diff
@@ -109,6 +109,46 @@ public void testEnumAsBindParameterAndExtract() {
 		} );
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HHH-10282")
+	public void hqlTestEnumShortHandSyntax() {
+		doInHibernate( this::sessionFactory, session -> {
+			session.createQuery(
+				"select id from Person where originalHairColor = BLONDE")
+				.getResultList();
+		} );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10282")
+	public void hqlTestEnumQualifiedShortHandSyntax() {
+		doInHibernate( this::sessionFactory, session -> {
+			session.createQuery(
+					"select id from Person where originalHairColor = HairColor.BLONDE")
+					.getResultList();
+		} );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10282")
+	public void hqlTestEnumShortHandSyntaxInPredicate() {
+		doInHibernate( this::sessionFactory, session -> {
+			session.createQuery(
+					"select id from Person where originalHairColor in (BLONDE, BROWN)")
+					.getResultList();
+		} );
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10282")
+	public void hqlTestEnumQualifiedShortHandSyntaxInPredicate() {
+		doInHibernate( this::sessionFactory, session -> {
+			session.createQuery(
+					"select id from Person where originalHairColor in (HairColor.BLONDE, HairColor.BROWN)")
+					.getResultList();
+		} );
+	}
+
```

Comment:
I think it would be ideal to include negation counterparts for each of the new testing cases above.